### PR TITLE
New Website Theme

### PR DESCRIPTION
**Pull request by waht**
# Purpose

New website theme with updated logo, new fonts, colors and minor changes to the start page.
Here is a preview:
theme

# Checklist

- [x] commit messages are fine (with references to issues)
- [x] I did local tests and everything went fine
- [x] release notes are updated (doc/news/_preparation_next_release.md)

@markus2330 please review my pull request

### CODE REVIEW

File: doc/news/_preparation_next_release.md [hunk 1]
```diff
@@ -34,22 +34,25 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.8.
 
 ## Highlights
 
-- New Logo
+- New Logo and Website Theme
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md]: At the moment it even looks like we can make an extra highlight for this.

File: doc/news/_preparation_next_release.md [hunk 2]
```diff
@@ -34,22 +34,25 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.8.
...
-We are proud to present our new Logo.
+We are proud to present our new logo.
 It has a new shape and cooler colors.
 
 <img src="https://cdn.rawgit.com/ElektraInitiative/libelektra/master/doc/images/logo/logo_color.svg" alt="Elektra" width="150" />
 
 Thanks to Philipp Frei
 
+We also gave the website a new look. It has the colors from the logo, new
+fonts for better readability and the start page structure was improved.
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md]: Which font? Please add link to Website here.

File: src/tools/rest-frontend/resources/assets/html/index.html.in [hunk 3]
```diff
@@ -22,6 +22,9 @@
         <!-- RSS Info -->
         <link rel="alternate" type="application/atom+xml" href="news/feed.rss" title="Elektra RSS Feed">
 
+        <!-- Fonts -->
+        <link href="https://fonts.googleapis.com/css?family=Lato|Libre+Franklin" rel="stylesheet">
```
Comment (markus2330) [inline src/tools/rest-frontend/resources/assets/html/index.html.in]: Please do not use cross-domain requests but copy the font into the repo (As done for other fonts already.).

Maybe some fonts are now obsolete and can be removed?

File: src/tools/rest-frontend/resources/assets/skin/bootstrap/list-group.less [hunk 4]
```diff
@@ -0,0 +1,6 @@
+.list-group-item.active {
+  &, &:focus, &:hover {
+    background-image: none;
+    box-shadow: none;
+  }
+}
```
Comment (markus2330) [inline src/tools/rest-frontend/resources/assets/skin/bootstrap/list-group.less]: Every text file should end with a new-line.

File: src/tools/rest-frontend/resources/assets/skin/bootstrap/navbar.less [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  box-shadow: none;
+  
+  .navbar-nav>.active>a,
+  .navbar-nav>.open>a {
+    background-image: none;
+    box-shadow: none;
+  }
+ }
+ 
+
```
Comment (markus2330) [inline src/tools/rest-frontend/resources/assets/skin/bootstrap/navbar.less]: Space?

**Comment by markus2330** [comment]
jenkins build homepage please

**Comment by markus2330** [comment]
The new website is online, I love it.

I noticed that the colors of the original buttons sometimes do not fit well, e.g. at https://www.libelektra.org/conversion but the buttons might be changed by @vanessaNWK anyway.

The font change is really minor, is it worth the trouble? Is it now the same font as in the logo?

File: src/tools/rest-frontend/resources/assets/html/index.html.in (hunk 3 above)
Comment (waht) [reply src/tools/rest-frontend/resources/assets/html/index.html.in]: Sure, this was intended for testing the fonts. I would convert and integtrate the fonts when we agreed upon them.

**Comment by waht** [comment]
Thank you! Seems to work great!

Can you be more specific about the buttons at the conversion page? Is it the blue "conversion" button or the red "report" button?
conversion

No its not the font from the logo (that did not look good at all). Granted, the difference is not that big but it improves readability (at least on my system) and helps to build a unique look for the website.
I created some screenshots with and without the fonts for comparison:

- [home with fonts](https://user-images.githubusercontent.com/2136968/35418248-89dc6fa8-0231-11e8-89e2-2216b81ae536.png)
- [home without fonts](https://user-images.githubusercontent.com/2136968/35418249-89f5fb4e-0231-11e8-8ae8-1a7a9fbb9748.png)
- [docs with fonts](https://user-images.githubusercontent.com/2136968/35418250-8a123dea-0231-11e8-984a-52bd720f1fd5.png)
- [docs without fonts](https://user-images.githubusercontent.com/2136968/35418251-8a308f48-0231-11e8-9af3-678a8521b7ac.png)

**Comment by markus2330** [comment]
> Can you be more specific about the buttons at the conversion page? Is it the blue "conversion" button or the red "report" button?

If my eyes do not betray me both colors do not match with the color scheme. It is really minor, though. It is definitely not a necessity to change this. (even if they are mismatching colors, which I did not check with a tool nor a side-by-side comparison).

> home without fonts

Okay, my "home without fonts" looks completely different, seems to be an issue of which fonts are installed. Let us pick up this discussion in the meeting today.

**Comment by markus2330** [comment]
jenkins build homepage please

**Comment by markus2330** [comment]
I have triggered a rebuild so that we can see the new website in the meeting today.

Btw. the RSS symbol could also be in Elektra's orange. (Again, really minor)

**Comment by waht** [comment]
jenkins build homepage please

**Comment by markus2330** [comment]
Looks great now! There are some formatting issues (spaces in CSS files) and could you please add SPDX-License-Identifier: SIL (see [SPDX](https://spdx.org/licenses/) and [reuse](https://reuse.software/practices/2.0/)). SIL seems to be designed to be distributed in ways like this but maintainers need to know that we do so. (They are copyleft and not BSD.)

Did you find any font not in use anymore?

**Comment by waht** [comment]
Thank you! I see, I replaced the spaces with tabs in the style files. 

About the licensing: The font files are binary so I could not add the SPDX identifier - I opted for an [DEP-5/copyright](https://www.debian.org/doc/packaging-manuals/copyright-format/1.0) file to state the license for all font files at once. I also searched for further copyright notes in files and included those along with their licenses as well. 
SPDX would also recommend adding all license texts in a LICENSES directory, but I haven't done that yet.

I removed the symbol font "font-awesome" from our source in favor of an npm-package.

**Comment by markus2330** [comment]
Thank you for taking care of the licensing. In the current situation we would get a merge conflict when we merge master into our debian branch which already has a debian/copyright file. I am not aware of a standard of how upstream can provide a debian/copyright file. I think the easiest way would be that you merge it into our debian branch in the already existing [debian/copyright](https://github.com/ElektraInitiative/libelektra/blob/debian/debian/copyright) file. (A separate PR is needed for this.)

Is the npm package of font-awesome reliable? We already had quite often problems of npm packages that suddenly break. It also seems weird that some fonts are included and other are downloaded as package.

Otherwise I cannot wait to get this merged, the website improvement is fabulous.

**Comment by markus2330** [comment]
jenkins build homepage please

**Comment by waht** [comment]
I understand - the package [seems quite fine](https://packagequality.com/#?package=font-awesome), but I added font-awesome back to keep things consistent. I'm creating the PR for debian right now.

**Comment by waht** [comment]
jenkins build homepage please

**Comment by markus2330** [comment]
For me even the website packagequality.com does not look trustworthy :-)

If you think its better this way, we can try it. But please do this migration in a separate commit so that we can revert it if the package causes trouble.

**Comment by waht** [comment]
My intuition is that the package is fine. It has versions along regular releases of the font-awesome project and the publisher is a (one-time though) contributor to the project.

But as long the other fonts are not available via npm it's not worth the trouble, and things are better if they are consistent.

File: src/tools/rest-frontend/package.json [hunk 6]
```diff
@@ -36,6 +36,7 @@
         "bootstrap": "^3.3.7",
         "connect-modrewrite": "^0.9.0",
         "fs-extra": "^0.30.0",
+        "font-awesome": "^4.7.0",
```
Comment (markus2330) [inline src/tools/rest-frontend/package.json]: Why do we need this line if we also have the fonts included in the repo?

Comment (waht) [reply src/tools/rest-frontend/package.json]: Yeah - we don't :) I'll remove it.

**Comment by markus2330** [comment]
Thank you! I'll trigger the website build from master.

**Pull request merged**